File: ABC/ABC185/abc185_e.cpp
#include <bits/stdc++.h>
#define rep(i, n) for(ll i = 0; i < (n); i++)
using namespace std;
typedef long long ll;
const ll MAX = 1010;

ll n, m;
ll a[MAX], b[MAX];
ll dp[MAX][MAX];
ll solve(void)
{
	for (ll i = 0; i < n; i++)
	{
		for (ll j = 0; j < m; j++)
		{
			if (a[i] == b[j])
				dp[i + 1][j + 1] = max(dp[i + 1][j + 1], dp[i][j] + 1);
			else
				dp[i + 1][j + 1] = max(dp[i][j + 1], dp[i + 1][j]);
		}
	}
	return dp[n][m];
}

signed main(void)
{
	ll ans;

	cin >> n >> m;
	rep(i, n) cin >> a[i];
	rep(i, m) cin >> b[i];
	ans = abs(n - m);
	ans += min(n, m) - solve();
	cout << ans << endl;
	return 0;
}

File: ABC/ABC185/abc185_e.py
def solve(N, M, A, B):
    dp = [[0 for _ in range(M+1)] for _ in range(N+1)]
    for i in range(N+1):
        dp[i][0] = i
    for i in range(M+1):
        dp[0][i] = i
    for i in range(N):
        for j in range(M):
            if A[i] == B[j]:
                dp[i+1][j+1] = dp[i][j]
            else:
                dp[i+1][j+1] = dp[i][j] + 1
            dp[i+1][j+1] = min(dp[i+1][j+1], dp[i+1][j]+1)
            dp[i+1][j+1] = min(dp[i+1][j+1], dp[i][j+1]+1)
    ans = dp[N][M]
    return ans

if __name__ == '__main__':
    N, M = map(int, input().split())
    A = list(map(int, input().split()))
    B = list(map(int, input().split()))
    ans = solve(N, M, A, B)
    print(ans)
